core/resharding.go

```diff
@@ -132,22 +133,47 @@ func GetEpochFromBlockNumber(blockNumber uint64) uint64 {
 }
 
 // GetShardingStateFromBlockChain will retrieve random seed and shard map from beacon chain for given a epoch
-func GetShardingStateFromBlockChain(bc *BlockChain, epoch uint64) *ShardingState {
+func GetShardingStateFromBlockChain(bc *BlockChain, epoch uint64,
+) (*ShardingState, error) {
 	number := GetBlockNumberFromEpoch(epoch)
-	shardState := bc.GetShardStateByNumber(number)
+	shardState, err := bc.GetShardStateByNumber(number)
+	if err != nil {
+		return nil, ctxerror.New("cannot get shard state for epoch",
+			"epoch", epoch).WithCause(err)
+	}
 
 	rndSeedBytes := bc.GetRandSeedByNumber(number)
+	if rndSeedBytes == [32]byte{} {
+		utils.GetLogInstance().Warn("cannot get randomness from block",
+			"blockNumber", number)
+		// TODO ek – should we fail here?  Continuing here would end up using
+		//  all-zeros for randomness, which seems like a security breach.
```

Comment:
@rlan35 what do you think?

Reply:
sure. sounds good. Leave it as TODO for now.